R/net.mod.init.R

```diff
@@ -136,26 +136,39 @@ initialize.net <- function(x, param, init, control, s) {
 #'
 init_status.net <- function(dat) {
 
+  type <- get_control(dat, "type", override.null.error = TRUE)
+  nsteps <- get_control(dat, "nsteps")
+  tgl <- get_control(dat, "tergmLite")
+  vital <- get_param(dat, "vital")
+  groups <- get_param(dat, "groups")
+  if (type %in% c("SIS", "SIR")){
+    rec.rate <- get_param(dat, "rec.rate")
+  }
+  if (vital) {
+    di.rate <- get_param(dat, "di.rate")
+  }
+
   # Variables ---------------------------------------------------------------
-  i.num <- dat$init$i.num
-  i.num.g2 <- dat$init$i.num.g2
-  r.num <- dat$init$r.num
-  r.num.g2 <- dat$init$r.num.g2
+  i.num <- get_init(dat, "i.num", override.null.error = TRUE)
+  if (type  == "SIR"){
+    r.num <- get_init(dat, "r.num", override.null.error = TRUE)
+  }
```

Comment:
shouldn't that NULL be caught with crosscheck?

Reply:
Subsequent change to status.vector to address this reverted. Updated and fixed.